src/MediatR/Composite/CompositeRequestHandler.cs

```diff
@@ -0,0 +1,60 @@
+namespace MediatR.Composite
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using System.Dynamic;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// defines an asynchronous handler for AsyncCompositeRequest <see cref="AsyncCompositeRequest"/>
+    /// </summary>
+    /// <remarks>Output will always be an ExpandoObject <see cref="ExpandoObject"/></remarks>
+    public class AsyncCompositeRequestHandler : IRequestHandler<AsyncCompositeRequest, ExpandoObject>
+    {
+        private IMediator _mediator;
+
+        public AsyncCompositeRequestHandler(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
+        public async Task<ExpandoObject> Handle(AsyncCompositeRequest compositeRequest, CancellationToken cancellationToken)
+        {
+            IDictionary<string, object> response = new ExpandoObject();
+            foreach (var request in compositeRequest.Requests)
+            {
+                response[request.Key] = await _mediator.Send(request.Value);
+            }
+
+            return response as ExpandoObject;
+        }
+    }
+
+
+
+    /// <summary>
+    /// defines a synchronous handler for CompositeRequests <see cref="CompositeRequest"/>
+    /// </summary>
+    /// <remarks>Output will always be an ExpandoObject <see cref="ExpandoObject"/></remarks>
+    public class CompositeRequestHandler : IRequestHandler<CompositeRequest, ExpandoObject>
+    {
+        private IMediator _mediator;
+
+        public CompositeRequestHandler(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
+        public Task<ExpandoObject> Handle(CompositeRequest compositeRequest, CancellationToken cancellationToken)
+        {
+            IDictionary<string, object> response = new ExpandoObject();
+            foreach (var request in compositeRequest.Requests)
+            {
+                response[request.Key] = _mediator.Send(request.Value);
```

Comment:
I'm not sure I understand the point of the `CompositeRequestHandler`.  All mediator handlers are async by default, so in this case aren't you just storing the tasks in the `response[request.Key]`, which are going to continue running as your loop through the requests?  In your tests you are only using `AsyncCompositeRequest`.

Reply:
No you're 100% correct. That was a hold over from an attempting synchronous method that never worked like I wanted it to (was to be inline the synchronous RequestHandler). I'll remove it.